### PR TITLE
Feature/client status

### PR DESCRIPTION
**Pull request by loneil**
# Description
Prerequesite work for:
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-654

Add an API endpoint to get KC client details in an SSO env for a acronym.
See https://getok-pr-111-k8vopl-dev.pathfinder.gov.bc.ca/api/v1/docs#operation/kcClientGet for spec proposal

## Types of changes

 New feature (non-breaking change which adds functionality) 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

Also adding a missing fix for KC service client life-cycle auditing

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-111

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-111

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-111

File: frontend/src/components/webAdeVisualizer/SecurityUtils.vue [hunk 1]
```diff
@@ -95,7 +95,9 @@
                     </tbody>
                   </template>
                 </v-simple-table>
-                <p v-if="insecurePasswordsList && insecurePasswordsList.length == 0">No results could be found for this search.</p>
+                <p
+                  v-if="insecurePasswordsList && insecurePasswordsList.length == 0"
```
Comment (jujaga) [inline frontend/src/components/webAdeVisualizer/SecurityUtils.vue]: v-if="insecurePasswordsList && !insecurePasswordsList.length"

File: backend/src/components/keyCloakServiceClientMgr.js [hunk 2]
```diff
@@ -98,6 +100,60 @@ class KeyCloakServiceClientManager {
       oidcTokenUrl: this.svc.tokenUrl
     };
   }
+
+  async fetchClient(applicationAcronym) {
+    log.info('KeyCloakServiceClientManager.fetchClient ', applicationAcronym);
```
[the remark this answers is not in the document]
Comment (loneil) [reply backend/src/components/keyCloakServiceClientMgr.js]: Fixed

File: backend/src/components/keyCloakServiceClientMgr.js [hunk 3]
```diff
@@ -98,6 +100,60 @@ class KeyCloakServiceClientManager {
...
+    if (!applicationAcronym) {
+      log.error('KeyCloakServiceClientManager.fetchClient - no applicationAcronym provided.');
```
[the remark this answers is not in the document]
Comment (loneil) [reply backend/src/components/keyCloakServiceClientMgr.js]: Fixed

File: backend/src/components/keyCloakServiceClientMgr.js [hunk 4]
```diff
@@ -98,6 +100,60 @@ class KeyCloakServiceClientManager {
...
+        id: serviceClient.id,
+        clientId: serviceClient.clientId,
+        enabled: serviceClient.enabled,
+        name: serviceClient.name,
+        description: serviceClient.description,
+        serviceAccountEmail: serviceAccountUser ? serviceAccountUser.email : ''
+      };
+      if (roleComposites && roleComposites.length) {
+        detailObject.commonServiceRoles = roleComposites
+          .filter(role => role.name != 'uma_protection')
```
[the remark this answers is not in the document]
Comment (loneil) [reply backend/src/components/keyCloakServiceClientMgr.js]: Fixed

File: backend/src/components/realmAdminSvc.js [hunk 5]
```diff
@@ -339,6 +339,26 @@ class RealmAdminService {
     return response.data;
   }
 
+  async getRoleComposites(clientId, roleName) {
+    if (!clientId) {
+      log.error('RealmAdminService getRoleComposites clientId parameter is null.');
```
[the remark this answers is not in the document]
Comment (loneil) [reply backend/src/components/realmAdminSvc.js]: Fixed

File: backend/src/components/realmAdminSvc.js [hunk 6]
```diff
@@ -339,6 +339,26 @@ class RealmAdminService {
...
+      throw new Error('Cannot get role composites for client roles: clientId parameter cannot be null.');
+    }
+    if (!roleName) {
+      log.error('RealmAdminService getRoleComposites roleName parameter is null.');
```
[the remark this answers is not in the document]
Comment (loneil) [reply backend/src/components/realmAdminSvc.js]: Fixed

File: backend/src/docs/v1.api-spec.yaml [hunk 7]
```diff
@@ -567,6 +603,48 @@ components:
...
+          type: string
+          example: This application does some work
+        serviceAccountEmail:
+          type: string
+          example: bob.alice@gov.bc.ca
+        commonServiceRoles:
+          type: array
+          items:
+            $ref: '#/components/schemas/Role'
+          description: An array of Common Service roles the client currently has
```
[the remark this answers is not in the document]
Comment (loneil) [reply backend/src/docs/v1.api-spec.yaml]: Will leave this one cause it has a distinct reason

File: backend/src/docs/v1.api-spec.yaml [hunk 8]
```diff
@@ -626,6 +704,18 @@ components:
           type: string
           description: Contains the service client name that was written to WebADE
           example: CHES_SERVICE_CLIENT
+    Role:
```
[the remark this answers is not in the document]
Comment (loneil) [reply backend/src/docs/v1.api-spec.yaml]: Adjusting to AccessRole

File: backend/src/routes/v1/keyCloak.js [hunk 9]
```diff
@@ -62,4 +69,45 @@ keyCloak.post('/configForm', [
...
+    }
+  } catch (error) {
+    log.error(error);
+    res.status(500).json({
+      message: error.message
+    });
+    return res;
+  }
+});
+
```
[the remark this answers is not in the document]
Comment (loneil) [reply backend/src/routes/v1/keyCloak.js]: Fixed

File: frontend/src/components/webAdeVisualizer/SecurityUtils.vue (hunk 1 above)
Comment (loneil) [reply frontend/src/components/webAdeVisualizer/SecurityUtils.vue]: Fixed

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-111

**Pull request merged**